File: packages/ios/Sources/CivUI/CivSupportResources.swift
// CivUI — CivSupportResources for SwiftUI
// Crisis and support contact links for sensitive forms.

import SwiftUI

/// Support resources callout for sensitive forms.
///
/// Renders an aside with a heading and slotted action links for
/// crisis hotlines, emails, etc. Mirrors the web `civ-support-resources` component.
public struct CivSupportResources<Content: View>: View {
    /// Heading text.
    public var heading: String
    /// Semantic heading level (2–6).
    public var headingLevel: Int
    /// Visual tone: "default" or "crisis".
    public var tone: String
    /// Slotted content (action links).
    public var content: () -> Content

    public init(
        heading: String = "",
        headingLevel: Int = 3,
        tone: String = "default",
        @ViewBuilder content: @escaping () -> Content = { EmptyView() }
    ) {
        self.heading = heading
        self.headingLevel = headingLevel
        self.tone = tone
        self.content = content
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Text(heading).bold()
            content()
        }
        .accessibilityElement(children: .contain)
        .accessibilityLabel(heading)
    }
}

File: packages/android/src/main/kotlin/gov/civui/components/CivSupportResources.kt
// CivUI — CivSupportResources for Jetpack Compose
// Crisis and support contact links for sensitive forms.

package gov.civui.components

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp

/**
 * Support resources callout for sensitive forms.
 *
 * Renders a heading with slotted action links for crisis hotlines,
 * emails, etc. Mirrors the web `civ-support-resources` component.
 */
@Composable
fun CivSupportResources(
    heading: String = "",
    headingLevel: Int = 3,
    tone: String = "default",
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit = {},
) {
    Column(modifier = modifier) {
        if (heading.isNotEmpty()) {
            Text(text = heading, fontWeight = FontWeight.Bold)
            Spacer(modifier = Modifier.height(8.dp))
        }
        content()
    }
}
